static/src/javascripts/projects/common/modules/commercial/user-features.js

```diff
@@ -147,14 +150,64 @@ const isPayingMember = (): boolean =>
     // If the user is logged in, but has no cookie yet, play it safe and assume they're a paying user
     isUserLoggedIn() && getCookie(PAYING_MEMBER_COOKIE) !== 'false';
 
-const lastContributionDate = getCookie('gu.contributions.contrib-timestamp');
+// number returned is Epoch time in milliseconds.
+// null value signifies no last contribution date.
+const getLastOneOffContributionDate = (): ?number => {
+    const cookie = getCookie(SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE);
+
+    if (!cookie) {
+        return null;
+    }
 
-const daysSinceLastContribution = daysSince(lastContributionDate);
+    // Contributions frontend set date time of last contribution using ISO 8601 format.
+    // Support frontend set date time of last contribution in Epoch milliseconds.
+    // If we first attempted to parse cookie in Epoch milliseconds format (parseInt()),
+    // then a value in ISO 8601 format would parse (incorrectly) e.g. 2018-08-17T16:11:10Z => 2018
+    // So first attempt to parse cookie in ISO 8601 format.
+
+    const parseIso = (c: string) => {
+        try {
+            const ms = Date.parse(c);
```

Comment:
Definitely a good call to be defensive, but `Date.parse()` and `parseInt()` don't throw exceptions (unlike `JSON.parse()`!) so it's OK to ditch the try/catch